app/services/jwt_token.rb

```diff
@@ -34,8 +34,17 @@ def to_s
   end
 
   # This method does not verify if the token signature is valid
+  # defined? keyword checks if expression is currently defined. It
+  # therefore, helps to memoize the nil JWT values.
   def decoded_payload
-    @decoded_payload ||= JWT.decode(token, nil, false).first
+    return @decoded_payload if defined?(@decoded_payload)
+    @decoded_payload = JWT.decode(token, nil, false).first
+
+    return nil unless @decoded_payload.is_a?(Hash)
```

Comment:
should we log something about this and also set `@decoded_payload` to `nil` so additional calls won't decode again?

Reply:
Done.